### PR TITLE
DCOS-21440: keep VIP label

### PR DESCRIPTION
**Pull request by nLight**
This PR fixes the way the UI supports VIPs. 

UI used to be very opinionated about the way VIPs should be. It would require a VIP to be `VIP_<enpoint_index>` but that's not the way backend expects them. Backend allows anything that starts with `vip|VIP` to be a valid VIP label. Because of that we need to keep vip label that we're getting from JSON.

## Testing

Please create an app with a few Service Endpoints and override VIP_* to anything valid:
- vipBla
- VIPblop
- VIP_129312

Please check different network modes along with single and multi container.

NB! `Vip` `vIp` etc are invalid.

Closes DCOS-21440

**Checklist**
- [x] Did you add a JIRA issue in a commit message or as part of the branch name?
- [ ] Did you add new unit tests?
- [ ] Did you add new integration tests?
- [ ] If this is a regression, did you write a test to catch this in the future?

### CODE REVIEW

File: plugins/services/src/js/reducers/serviceForm/PortMappings.js [hunk 1]
```diff
@@ -119,19 +119,28 @@ module.exports = {
         });
       }
 
-      const vip = findNestedPropertyInObject(item, `labels.VIP_${index}`);
+      const vip = Object.entries(item.labels || {}).find(([key, _]) =>
```
Comment (DanielMSchmidt) [inline plugins/services/src/js/reducers/serviceForm/PortMappings.js]: I see this `.find(([key, _]) => key.match(/^(vip|VIP)/)` a third time and it is a bit cryptic to me, which is normal with regexes I guess. Can we move it into a util, I think there already is one for labels

Comment (jaxesn) [reply plugins/services/src/js/reducers/serviceForm/PortMappings.js]: any reason not to just use a case insensitive search? `key.match(/^vip/i)`

File: plugins/services/src/js/reducers/serviceForm/Container.js [hunk 2]
```diff
@@ -151,7 +151,7 @@ const containerJSONReducer = combineReducers({
 
     // Convert portDefinitions to portMappings
     return this.portDefinitions.map((portDefinition, index) => {
-      const vipLabel = `VIP_${index}`;
+      const vipLabel = portDefinition.vipLabel || `VIP_${index}`;
```
Comment (jaxesn) [inline plugins/services/src/js/reducers/serviceForm/Container.js]: Probably not super important, but the default label shows up a few places `VIP_${index}`, worth making that a constant/helper somewhere?

File: plugins/services/src/js/reducers/serviceForm/PortMappings.js (hunk 1 above)
Comment (DanielMSchmidt) [reply plugins/services/src/js/reducers/serviceForm/PortMappings.js]: Yeah, I read somewhere that VIP and vip is okay, Vip or vIP not

Comment (jaxesn) [reply plugins/services/src/js/reducers/serviceForm/PortMappings.js]: makes sense, thanks

Comment (nLight) [reply plugins/services/src/js/reducers/serviceForm/PortMappings.js]: @DanielMSchmidt ✅

File: plugins/services/src/js/reducers/serviceForm/Container.js (hunk 2 above)
Comment (nLight) [reply plugins/services/src/js/reducers/serviceForm/Container.js]: It was a bug fix PR, I added that refactoring, please test thoroughly.

**Comment by nLight** [comment]
@DanielMSchmidt please give it another shot!

**Comment by nLight** [comment]
@jaxesn @DanielMSchmidt refactoring has been addressed, please retest.

**Comment by nLight** [comment]
Hold merge until re-tested re-reviewed

**Comment by juliangieseke** [comment]
hm, github hasnt dismissed the reviews? why didnt you manually dismiss them? 🤔

**Comment by DanielMSchmidt** [comment]
@nLight re-reviewed it 👍

**Pull request merged**